### PR TITLE
Avoid preprocessor define.

### PR DESCRIPTION
**Pull request by bangerth**
What we perform is a "none of" operation. We can put all of that into `std::none_of` (which has the added benefit of documenting what we are doing) and then the `std::none_of` call can be right inside `Assert` and we no longer need the `#ifdef...#endif` part.

### CODE REVIEW

File: include/deal.II/lac/affine_constraints.templates.h [hunk 1]
```diff
@@ -901,20 +901,28 @@ AffineConstraints<number>::close()
...
+                              // make sure that entry->first is not the index of
+                              // a line itself
+                              const bool is_circle =
+                                is_constrained(entry.first);
+                              if (is_circle)
+                                return true;
+                            }
+                        return false;
+                      }),
+         ExcMessage("The constraints represented by this object have a cycle."
```
Comment (drwells) [inline include/deal.II/lac/affine_constraints.templates.h:924]:
```suggestion
         ExcMessage("The constraints represented by this object have a cycle. "
```

**Review by drwells — COMMENTED** [review]
One minor comment.

**Review by masterleinad — APPROVED** [review]

**Comment by bangerth** [comment]
Fixed as requested.

**Review by tamiko — APPROVED** [review]

**Comment by bangerth** [comment]
Tests cleanly.

**Pull request merged**